packages/devtools_app/lib/src/debugger/flutter/controls.dart

```diff
@@ -21,88 +21,97 @@ class DebuggingControls extends StatelessWidget {
   Widget build(BuildContext context) {
     return ValueListenableBuilder(
       valueListenable: controller.isPaused,
-      builder: (context, isPaused, child) {
-        final canStep = isPaused && controller.hasFrames.value;
-
-        return SizedBox(
-          height: Theme.of(context).buttonTheme.height,
-          child: Row(
-            children: [
-              RoundedOutlinedBorder(
-                child: Row(
-                  children: [
-                    DebuggerButton(
-                      title: 'Pause',
-                      icon: Icons.pause,
-                      autofocus: true,
-                      onPressed: isPaused ? null : controller.pause,
+      builder: (context, isPaused, _) {
+        return ValueListenableBuilder(
```

Comment:
another option would be to make DebuggingControls a stateful widget and then add listeners that call setState when either of these value listenables change, but I'm not sure that is better than what you already have.

You could break the content up into a helper method if you think that makes it more clear. e.g.

```
return ValueListenableBuilder(
  ...
    return ValueListenableBuilder(
    ...
      return _buildControls(paused: isPaused, maybeResuming: maybeResuming);
    ),
);
```

Reply:
Yeah, likely not something to optimize now, but perhaps if we really thought we could improve the code readability we could have a widget line `ValueListenableBuilder2<T, S>` that could handle two ValueListeners.